### DISCUSSION TITLE
Rename Node.name to Node.node_name

### DISCUSSION THREAD

**Discussion by zhehangd** [godotengine/godot-proposals#10859] [Scripting]
It comes too often that I want to define a property called "name", for example name of a character but have to use a different identifier to avoid conflict. For example, to use "character_name". It is ok, just a bit inconvenient. The key problem is that it is so easy to write error-prone code.

When I want to change the name of the character, I should write

```gdscript
character.character_name = "foo"
```

But I find myself frequently write

```gdscript
character.name = "foo"
```

It ends up with no editor or runtime error as it actually changes the name of the node, so this bug can be hard to find. Since "name" is such a common name and every node has to inherit from Node. I think it might be better to let Node not occupy this identifier and use a different one, such as "node_name" (as well as "set_node_name", "get_node_name").

**Comment by Calinou** [comment]
If we rename the property, we'd have to keep the old one around for backwards compatibility, so `name` would remain reserved anyway.

**Comment by Mickeon** [comment]
I've gotten kind of used to this but I get where the issue would arise. The **Resource** class preemptively prefixed its properties with `resource_` to avoid this issue, but it's not as easy to do with Node.

Not just for the caveat Calinou described, but also because there's could be few more properties to go through and rename as well. For example, I personally have to admit that the **Node**'s `owner` stepped on my toes for the same reasons highlighted on top.

As a point of reference, **CanvasItem**'s `update()` method in 3.x was renamed to `queue_redraw()` ([by me](https://github.com/godotengine/godot/pull/64377)) because, mainly, the original purpose of `update` is much vaguer and misleading in the inheritance chain. "What does `update()` do on a Skeleton2D?"

And, overall, one could argue this to be the case for this property, as well.
But regardless, changing `name`'s... name would prove problematic for users that did, in fact, get used to it, no matter the final result.

**Comment by azur-wolve** [comment]
this has been requested and discussed multiple times

the problem is that has been since first Godot version, so all code that is already out there would break

I agree that using `Node.node_name` would be better
